### ISSUE TITLE
Hot Restart Button on Flutter Debug mode,  does not work as expected.

### ISSUE THREAD

**Issue by samuelematias** [emacs-lsp/lsp-dart#97]
**Describe the bug**
On Flutter Debug mode, the `Hot Restart Button`, does not work as expected.

**Flutter Debug  Helper Palette**
![Screen Shot 2021-04-08 at 17 56 14](https://user-images.githubusercontent.com/5155386/114095800-424f3100-9894-11eb-92ba-699728b3377a.png)

**The `Hot Restart Button`**
![Screen Shot 2021-04-08 at 17 56 18](https://user-images.githubusercontent.com/5155386/114095835-4c712f80-9894-11eb-8b32-6a81d98ba01a.png)

When clicked, the application closes.

![hot-restart-button-bug](https://user-images.githubusercontent.com/5155386/114096016-8e9a7100-9894-11eb-8e7c-3a4cb97248d5.gif)

Where it should actually perform the hot restart of the app, which would be the same as when using the `lsp-dart-dap-flutter-hot-restart `function.

![hot-restart-button-need-do](https://user-images.githubusercontent.com/5155386/114095949-74609300-9894-11eb-97c6-970ab944acc5.gif)

**To Reproduce**
`M-x`  -> `dap-debug` -> `Flutter :: Debug`, with the app running and the`Flutter Debug  Helper Palette` showing, click on the `Hot Restart Button` (the Green Button).

**Expected behavior**
`Hot Restart` the app.

**Actual behavior**
The app closes.

**Version**
![Screen Shot 2021-04-08 at 17 54 35](https://user-images.githubusercontent.com/5155386/114096512-2e57ff00-9895-11eb-9de6-b0a5b6ca646e.png)

**Comment by ericdallo** [comment]
Thanks for the report!

Actually that button is generic for `dap-mode` and that button means `Restart the debug session` 😅 
@yyoncho maybe we could allow changing that or introduce some other entrypoint?

**Comment by elken** [comment]
> Thanks for the report!
> 
> Actually that button is generic for `dap-mode` and that button means `Restart the debug session` 😅 @yyoncho maybe we could allow changing that or introduce some other entrypoint?

Just to clarify; this is the behaviour on vscode (the button does a hot restart not a full restart). Seems like there could be a `dap-ui-restart-button-function` or something to decide what to use to "restart" the session?

**Comment by ericdallo** [comment]
yes, ATM the dap-ui-control is generic for all languages, and dart/flutter is the only one that we want this behavior, so we need to make dap-mode accept this custom behavior change c/c @yyoncho

**Comment by yyoncho** [comment]
I think that restart implementation in dap-mode is incorrect and that it should somehow be generic and handle that case. But adding extension points is generally fine.
